TheMovieDb/Modules/Detail/DetailView.swift

```diff
@@ -105,6 +103,16 @@ final class DetailView: UIViewController {
         infoContainer.addArrangedSubview(similarMovies)
         infoContainer.addArrangedSubview(recommendationMovies)
         infoContainer.addArrangedSubview(movieCast)
+        if let posterpath = viewModel.getMoviePosterPath(),
+           let posterURL = URL(string: MovieDBAPI.APIConstants.imageUrl + posterpath) {
+            if let cached = cache[posterURL.absoluteString] {
+                poster.image = cached
+            } else {
+                poster.load(url: posterURL) { [weak self] image in
+                    self?.cache[posterURL.absoluteString] = image
+                }
+            }
+        }
```

Comment:
🤔  in this case you can handle this logic in your ViewModel and your ViewController only accepts an image